.github/workflows/release.yml

```diff
@@ -48,7 +48,7 @@ jobs:
         run: ./gradlew assemble
       - name: "📤 Upload Distribution"
         if: success()
-        uses: actions/upload-artifact@v4
+        uses: actions/upload-artifact@50769540e7f4bd5e21e526ee35c689e35e0d6874 # v4.4.0
```

Comment:
I was talking to @jamesfredley  about this.  I understand it's trusted so we don't have to worry about a supply chain attacks, but in my diffs I noticed some of these had the sha's.  I got curious and dug into the upload-artifact specifically: 

https://github.com/actions/upload-artifact/releases/tag/v4.4.0

Apparently they made a breaking change in v4.  I'm under the impression that it will always run with the latest without the sha, and if they're making breaking changes that affects our build reproducibility.  I'm assuming it's more important to be able to reproduce builds like they were when they first released so we probably should pin the sha's to ensure that.  

I had planned to bring this up in the stewards call.  If you want, I can revert this for now and we can discuss more as a group.  What are your thoughts?

Reply:
Yeah, that seems to have been a mistake not making that change in v5 and just issuing warnings in v3 and v4, (looking at the comments on issue https://github.com/actions/upload-artifact/issues/602).
This is of course a risk using major versions.

I don't have strong opinions about it, but we should be consistent.
Leave it in, and lets discuss how we should handle these versions going forward.

Reply:
Thank you.  I'll leave it in on these and we can discuss in the stewards meeting.